src/ostree/ot-remote-builtin-add.c

```diff
@@ -66,6 +69,20 @@ ot_remote_builtin_add (int argc, char **argv, GCancellable *cancellable, GError
                                     OSTREE_BUILTIN_FLAG_NONE, &repo, cancellable, error))
     goto out;
 
+  /* As a special case, we can take a --sysroot argument. Currently we also
+   * require --repo because fixing that needs more cmdline rework.
+   */
```

Comment:
Indeed, good idea.  It turned out to be somewhat more subtle to implement than I thought; my original version of the below :arrow_down: just returned an `OstreeRepo*` but in fact the way the repo holds a weak ref to the sysroot meant that weak ref went away (since we were only transiently holding the sysroot), and hence the "is physical" state (plus the sysroot fd) were no longer accessible.

Fixed that by holding onto the `OstreeSysroot` on the toplevel.  Perhaps what we should really do is just have every ostree command also accept `--sysroot` or so.  Maybe later.